timelock-agent/src/main/java/com/palantir/timelock/utils/KubernetesHostnames.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.utils;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.palantir.logsafe.SafeArg;
+
+public final class KubernetesHostnames {
+
+    public static final KubernetesHostnames INSTANCE = new KubernetesHostnames(() -> {
+        try {
+            return InetAddress.getLocalHost().getCanonicalHostName();
+        } catch (UnknownHostException e) {
+            throw new RuntimeException(e);
+        }
+    });
+
+    private static final Logger log = LoggerFactory.getLogger(KubernetesHostnames.class);
+    private static final String POD_HOSTNAME_TEMPLATE = "%s-%d.%s.%s";
+
+    private final Supplier<String> currentHostnameSupplier;
+
+    /** The pattern for hostnames when running in a k8s statefulset. */
+    private static final Pattern POD_HOSTNAME_TEMPLATE_REGEX = Pattern.compile(
+            "(?<service>[a-z0-9\\-.]+)-(?<podId>\\d+)\\.\\k<service>"
+                    + "\\.(?<namespace>[a-z0-9\\-.]+)\\.svc\\.cluster\\.local");
```

Comment:
OK, confirmed this matches the regex we normally use for this kind of path hacking 👍